src/Pages/Profile/admin/AdminView.jsx

```diff
@@ -0,0 +1,294 @@
+        type: 'email',
+        placeholder: 'make it secure',
+        onChange: (e) => setDoorCode(e.target.value)
+      },
+      {
+        label: 'Major',
+        type: 'email',
+        placeholder: user.major,
+        onChange: (e) => setMajor(e.target.value)
+      }
+    ]
+
+    return (
+      <div>
+        <div>
+          <Button
+            color="primary"
+            style={{
+              position: 'relative',
+              left: '80%'
+            }}
+            onClick={() => { handleToggle() }}>
+            Edit
+          </Button>
+        </div>
+
+        <Modal isOpen={toggle}>
+          <ModalHeader>Modal title</ModalHeader>
+          <ModalBody>
+            <Form>
+
+              {FormGroups.map((group, index) => {
+                return <FormGroup key={index}>
+                  <Label>{group.label}</Label>
+                  <Input
+                    type={group.type}
+                    name={group.type}
+                    placeholder={group.placeholder}
+                    onChange={group.onChange}
+                  />
+                </FormGroup>
+              })}
+
+              Change validation date to
+              <select onChange={(e)=>{setNumberOfSemestersToSignUpFor(e.target.value)}}>
+                <option value={0}>Keep Same</option>
+                <option value={1}>This semester</option>
+                <option value={2}>2 semesters</option>
+              </select>
+
+              <Button
+                type='button'
+                onClick={() => {
+                  setPagesPrinted(0);
+                }}
+                color="info"
+                style={{marginTop:'5px'}}>
+                Reset Pages!
+              </Button>
+
+              <FormGroup tag="fieldset">
+                <legend>Membership Status</legend>
+                {enums.getAllValues(enums.membershipStatus).map((membership, index) => {
+                  return <FormGroup check key={index}>
+                    <Label check>
+                      <Input
+                        type="radio"
+                        name="radio1"
+                        value={membership}
+                        onChange={() => {
+                          setuserMembership(membership)
+                        }}
+                      />
+                      {enums.getKey(enums.membershipStatus,membership)}
+                    </Label>
+                  </FormGroup>
+                })}
+              </FormGroup>
+
+            </Form>
+          </ModalBody>
+
+          <ModalFooter>
+            <Button
+              color="primary"
+              onClick={() => { handleSubmissionToggle() }}>
+              Submit
+            </Button>
+            <Button
+              color="secondary"
+              onClick={() => { handleToggle() }}>
+              Cancel
+            </Button>
+          </ModalFooter>
+
+        </Modal>
+      </div>
+    );
+  }
+
+  return (
```

Comment:
this return block should also be its own react component.

i understand that refactoring this is cumbersome, but we are going for maintainability. last thing i'd want is all our hard work to get deprecated.